### DISCUSSION TITLE
The correct way to use `FeedView` with C # markup?

### DISCUSSION THREAD

**Discussion by qhj** [unoplatform/uno#18333] [Q&A] [answered]
**1. can't use binding on `RefreshContainerExtensions`**

![image](https://github.com/user-attachments/assets/41691798-81cd-4374-9b86-7d0efcc04c45)
---
![image](https://github.com/user-attachments/assets/51264f76-7369-4ffa-a861-d005b71ea65f)

**2. `ValueTemplate` and `IFeed`**

![image](https://github.com/user-attachments/assets/9a2685ed-e5cc-462f-8b33-166211590486)

**3. display `Excapetion` message in `ErrorTemplate`**

**Comment by MartinZikmund** [comment]
CC @dr1rrb

**Comment by bitbltr** [comment, marked as answer]
I struggled and finally figured it out. I don't remember where I saw the inspiration, if it was here or Stack Overflow. Something like this:
```
new FeedView()
    .Source(() => vm.MyDataSource)
    .ValueTemplate<FeedViewState>(feedViewState =>
        new ListView()
            .ItemsSource(() => feedViewState.Data!)
            .ItemTemplate<MyItemType>(item => new TextBlock().Text(() => item.SomeStringValue)))
```
FeedViewState is internal to FeedView and the .Data property is your data source

**Reply by MartinZikmund** [reply]
Thank you for providing the solution, sorry we didn't manage to get to it earlier 😑
